mkosi/backend.py

```diff
@@ -634,6 +639,24 @@ def get_partition(self, ident: PartitionIdentifier) -> Optional[Partition]:
             return None
         return self.partition_table.partitions.get(ident)
 
+    @property
+    def installer(self) -> Optional["DistributionInstaller"]:
+        try:
+            # we do a getattr here because we wnt to use the missing attribute
+            # as a sentinel, since we will use None as a sentinel further down
+            # for the case of distros that are not migrated to this interface
+            return cast(Optional["DistributionInstaller"], getattr(self, "_installer"))
+        except AttributeError:
+            try:
+                distro = str(self.config.distribution)
+                mod = importlib.import_module(f"mkosi.distributions.{distro}")
+                installer = getattr(mod, f"{distro.capitalize()}Installer")
+                instance = installer() if issubclass(installer, DistributionInstaller) else None
+            except (ImportError, AttributeError):
+                instance = None
+            setattr(self, "_installer", instance)
+            return cast(Optional["DistributionInstaller"], instance)
```

Comment:
Yes, this *is* unnecessarily complex, but it's only scaffolding. I wanted to port some more distros this weekend and once all of them are ported this can go away pretty much wholesale.

Reply:
It's bit less complex now, since the extra sentinel is gone, now we just dynamically load the the installer from the module. This is necessary, because otherwise there would be an import loop, since the distribution modules always load at least `MkosiState` from `backend.py`.